portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,295 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
```

Comment:
I think unit tests should be in the same package as the SUT. It makes it easier to access the non-public methods for testing.
```suggestion
package com.google.sps.servlets;
```
Also don't forgetr to also move the file to the correct directory afterwards.
`portfolio/src/test/java/com/google/sps/servlets/DataServletTest.java`

Reply:
that worked great! thanks Kaveh!